### PR TITLE
[Bugfix][Support Struct]Fixed some expressions not supporting qualified name when creating views.

### LINKED ISSUE #10738: Some expressions dont support qualified name.
### Steps to reproduce the behavior (Required)

```sql
CREATE TABLE `tj` (
    `id` INT(11) NOT NULL COMMENT "",
    `j`  JSON NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`id`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`id`) BUCKETS 1
PROPERTIES (
    "replication_num" = "1",
    "in_memory" = "false",
    "storage_format" = "DEFAULT"
);
```
```sql
INSERT INTO tj (id, j) VALUES (1, parse_json('{"a": 1, "b": true}'));
```

```sql
CREATE VIEW test.example_view(x) AS SELECT tj.j->'a' FROM tj;
```

### Expected behavior (Required)

```bash
Query OK, 0 rows affected (0.03 sec)
```

### Real behavior (Required)

```bash
failed to init view stmt
```

### StarRocks version (Required)

`main c759c03`

### PR DESCRIPTION
**Pull request by Smith-Cruise**
The test case will first try to run the original SQL query, then the test case will run `create view` based on the previous query. If some expressions are using the qualified name, `create view` SQL will parse failed.

This bug we must fix it before we support dot expression in Struct.

## What type of PR is this：
- [x] bugfix
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #10738

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/ExpressionMapping.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SelectAnalyzer.java [hunk 1]
```diff
@@ -225,11 +225,14 @@ private List<Expr> analyzeSelect(SelectList selectList, Relation fromRelation, b
...
             } else {
-                String name = item.getAlias() == null ? AST2SQL.toString(item.getExpr()) : item.getAlias();
-
                 analyzeExpression(item.getExpr(), analyzeState, scope);
                 outputExpressionBuilder.add(item.getExpr());
 
+                // The item here has been analyzed, it will contain qualified name information.
+                // For example, If an item originally only has a column name, after analyzeExpression(),
+                // item will have a qualified name.
+                String name = item.getAlias() == null ? AST2SQL.toString(item.getExpr()) : item.getAlias();
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SelectAnalyzer.java:234]: Field is designed to store the Column name, which is not expected

**Comment by HangyuanLiu** [comment]
Need to override UT, grouping set to create view, output column name of grouping function

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/ExpressionMapping.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/ExpressionMapping.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|2|2|100.00%|[]|

**Review by HangyuanLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/ExpressionMapping.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|2|2|100.00%|[]|

**Review by DorianZheng — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Smith-Cruise** [comment]
run starrocks_admit_test

**Comment by Smith-Cruise** [comment]
run starrocks_admit_test

**Pull request merged**